dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/AsyncRpcResult.java

```diff
@@ -203,18 +201,51 @@ public void setAttachment(String key, String value) {
     private RpcContext tmpContext;
     private RpcContext tmpServerContext;
 
-    private Function<Result, Result> beforeContext = (result) -> {
+    private Function<Result, Result> beforeContext = (appResponse) -> {
         tmpContext = RpcContext.getContext();
         tmpServerContext = RpcContext.getServerContext();
         RpcContext.restoreContext(storedContext);
         RpcContext.restoreServerContext(storedServerContext);
-        return result;
+        return appResponse;
     };
 
-    private Function<Result, Result> afterContext = (result) -> {
+    private Function<Result, Result> afterContext = (appResponse) -> {
         RpcContext.restoreContext(tmpContext);
         RpcContext.restoreServerContext(tmpServerContext);
-        return result;
+        return appResponse;
     };
+
+    /**
+     * Some utility methods used to quickly generate default AsyncRpcResult instance.
+     */
+    public static AsyncRpcResult newDefaultAsyncResult(AppResponse appResponse, Invocation invocation) {
+        AsyncRpcResult asyncRpcResult = new AsyncRpcResult(invocation);
+        asyncRpcResult.complete(appResponse);
+        return asyncRpcResult;
+    }
+
+    public static AsyncRpcResult newDefaultAsyncResult(Invocation invocation) {
+        return newDefaultAsyncResult(null, null, invocation);
+    }
+
+    public static AsyncRpcResult newDefaultAsyncResult(Object value, Invocation invocation) {
+        return newDefaultAsyncResult(value, null, invocation);
+    }
+
+    public static AsyncRpcResult newDefaultAsyncResult(Throwable t, Invocation invocation) {
+        return newDefaultAsyncResult(null, t, invocation);
+    }
+
+    public static AsyncRpcResult newDefaultAsyncResult(Object value, Throwable t, Invocation invocation) {
```

Comment:
I think we should check whether exception should be invoked here. 

There exception include:  
1. Client Exception
2. Server Exception.
3. Timeout or Net Exception or Provider not Found Exception.